src/StimulusBundle/src/Form/Extension/FormTypeExtension.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Form\Extension;
+
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\UX\StimulusBundle\Dto\StimulusAttributes;
+use Twig\Environment;
+use Twig\Loader\ArrayLoader;
+
+class FormTypeExtension extends AbstractTypeExtension
+{
+    private StimulusAttributes $stimulusAttributes;
+
+    public static function getExtendedTypes(): iterable
+    {
+        return [FormType::class];
+    }
+
+    public function buildView(FormView $view, FormInterface $form, array $options): void
+    {
+        if (
+            isset($options['stimulus_controller'])
+            || !isset($options['stimulus_target'])
+            || !isset($options['stimulus_action'])
+        ) {
+            $this->stimulusAttributes = new StimulusAttributes(new Environment(new ArrayLoader()));
+
+            if (isset($options['stimulus_controller'])) {
+                $this->handleController($options['stimulus_controller']);
+            }
+
+            if (isset($options['stimulus_target'])) {
+                $this->handleTarget($options['stimulus_target']);
+            }
+
+            if (isset($options['stimulus_action'])) {
+                $this->handleAction($options['stimulus_action']);
+            }
+
+            $attributes = array_merge($view->vars['attr'], $this->stimulusAttributes->toArray());
+            $view->vars['attr'] = $attributes;
```

Comment:
```suggestion
            $view->vars['attr'] = [...$view->vars['attr'], ...$this->stimulusAttributes->toArray()];
```

Avoid unnecessary var